test/dhcp_isc/integration_test.rb

```diff
@@ -39,3 +40,22 @@ def test_features
     end
   end
 end
+
+class DhcpApiValidIPTest < Test::Unit::TestCase
+  include Rack::Test::Methods
+
+  def app
+    app = Proxy::DhcpApi.new
+    app.helpers.server = @server
+    app
+  end
+
+  def test_validate_ip
+    @server = mock()
+    @server.stubs(:unused_ip).returns(true)
+    @server.expects(:validate_supported_address).with("192.168.122.0").returns("192.168.122.0")
+    @server.expects(:validate_supported_address).with("192.168.122.10", "192.168.122.20").returns("192.168.122.20")
```

Comment:
I think this is not really an integration test. I'd suggest to not stub this.

It would also be a good test to perform a request with an invalid IP address.